### PR TITLE
Fake quant weights and activations

### PR DESCRIPTION
**Pull request by TonyCongqianWang**

### CODE REVIEW

**Comment by vondele** [comment]
cscs-ci run RECIPE=threats

**Comment by vondele** [comment]
cscs-ci run RECIPE=threats

**Comment by vondele** [comment]
cscs-ci run RECIPE=threats

**Comment by vondele** [comment]
@TonyCongqianWang can you resolve the conflicts / rebase on master so it can be merged.

**Comment by TonyCongqianWang** [comment]
Should we perhaps merge nnue-pytorch first in this case? Will save us the explicit update to master step later on.

The qat net that passed fishtest doesn't actually need any parameters different from the previous master net as the default configs in nnue-pytorch were modified for them. Thus in this case it's just a update to master 

Edit: after looking more closely. There is the new parameter of factorized weight decay. I can handle adding it

**Comment by vondele** [comment]
I'd rather just merge this first, the nnue-pytorch PR is probably the most complex of the three.

**Comment by TonyCongqianWang** [comment]
Done
